modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -116,9 +139,81 @@ public CompletableFuture<Object> invoke(ReplicaRequest request) {
             return processMultiEntryAction((ReadWriteMultiRowReplicaRequest) request);
         } else if (request instanceof ReadWriteSwapRowReplicaRequest) {
             return processTwoEntriesAction((ReadWriteSwapRowReplicaRequest) request);
+        } else if (request instanceof ReadWriteScanRetrieveBatchReplicaRequest) {
+            return processScanRetrieveBatchAction((ReadWriteScanRetrieveBatchReplicaRequest) request);
+        } else if (request instanceof ReadWriteScanCloseReplicaRequest) {
+            return processScanCloseAction((ReadWriteScanCloseReplicaRequest) request);
         } else {
-            throw  new UnsupportedReplicaRequestException(request.getClass());
+            throw new UnsupportedReplicaRequestException(request.getClass());
+        }
+    }
+
+    /**
+     * Close all cursors connected with a transaction.
+     *
+     * @param txId Transaction id.
+     * @throws Exception When an issue happens on cursor closing.
+     */
+    public void closeAllTransactionCursors(UUID txId) throws Exception {
+        var lowCursorId = new IgniteUuid(txId, Long.MIN_VALUE);
+        var upperCursorId = new IgniteUuid(txId, Long.MAX_VALUE);
+
+        Map<IgniteUuid, Cursor<BinaryRow>> txCursors = cursors.subMap(lowCursorId, true, upperCursorId, true);
+
+        for (Cursor<BinaryRow> cursor : txCursors.values()) {
+            cursor.close();
+        }
+
+        txCursors.clear();
+    }
+
+    /**
+     * Precesses scan close request.
+     *
+     * @param request Scan close request operation.
+     * @return Listener response.
+     */
+    private CompletableFuture<Object> processScanCloseAction(ReadWriteScanCloseReplicaRequest request) {
+        UUID txId = request.transactionId();
+
+        IgniteUuid cursorId = new IgniteUuid(txId, request.scanId());
+
+        Cursor<BinaryRow> cursor = cursors.remove(cursorId);
+
+        if (cursor != null) {
+            try {
+                cursor.close();
+            } catch (Exception e) {
+                return CompletableFuture.failedFuture(e);
+            }
         }
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Precesses scan retrieve batch request.
+     *
+     * @param request Scan retrieve batch request operation.
+     * @return Listener response.
+     */
+    private CompletableFuture<Object> processScanRetrieveBatchAction(ReadWriteScanRetrieveBatchReplicaRequest request) {
+        UUID txId = request.transactionId();
+        int batchCount = request.batchCounter();
+
+        IgniteUuid cursorId = new IgniteUuid(txId, request.scanId());
+
+        return lockManager.acquire(cursorId.globalId(), new LockKey(tableId), LockMode.SHARED).thenCompose(tblLock -> {
```

Comment:
Minor. Seems that request.transactionId() instead of cursorId.globalId() is a bit clearer here.

Reply:
Fixed.